lib/cos.js

```diff
@@ -0,0 +1,134 @@
+
+exports.initialize = function initializeDataSource(dataSource, callback) {
+	//console.log(dataSource.settings);
+	dataSource.connector = new COSConnector(dataSource.settings);
+	process.nextTick(function () {
+		callback && callback();
+	});
+};
+var Connector = require('loopback-connector').Connector;
+
+require('util').inherits(COSConnector, Connector);
+
+const IBM = require('ibm-cos-sdk');
+
+exports.COSConnector = COSConnector;
+
+function COSConnector(dataSourceProps) {
+
+	this.endpoint = dataSourceProps.endpoint;
+	this.apiKeyId = dataSourceProps.apiKeyId;
+	this.ibmAuthEndpoint = 'https://iam.ng.bluemix.net/oidc/token';
+	this.serviceInstanceId = dataSourceProps.serviceInstanceId;
+	this.bucket = dataSourceProps.bucket;
+	this.response = [];
+	this._models = {};
+
+	this.config = {
+		endpoint: this.endpoint,
+		apiKeyId: this.apiKeyId,
+		ibmAuthEndpoint: this.ibmAuthEndpoint,
+		serviceInstanceId: this.serviceInstanceId,
+	};
+	this.cos = new IBM.S3(this.config);
+
+}
+
+
+//creates entry in bucket
+COSConnector.prototype.create = function create(model, data, callback){
+
+	var key = guid();
+	var dataStr = JSON.stringify(data); 
+
+	return this.cos.putObject({
+		'Bucket': this.bucket, 
+		'Key': key, 
+		'Body': dataStr,
+	}).promise()
+		.then(() => {
+    
+			data.id=key;
+			this.response.push(data);
```

Comment:
You should not keep any request-specific state in the connector instance. When multiple requests are made in parallel, this instance-level state will contain data from all requests!

For example, if you `create` two models in parallel, then `this.response` will contain both responses. That does not seem correct to me.

Am I missing something special about COS API?